tests/ui/impl-trait/in-trait/specialization-broken.stderr

```diff
@@ -18,6 +18,14 @@ LL |     fn bar(&self) -> impl Sized;
    = note: expected signature `fn(&U) -> impl Sized`
```

Comment:
This fails to be fixed because we're not actually even assembling a `ProjectionCandidate::ImplTraitInTrait` here.